src/test/java/dev/jbang/util/TestUtil.java

```diff
@@ -158,4 +160,13 @@ void testDispostionFilename() {
 		// assertThat(Util.getDispositionFilename("inline;
 		// filename*=iso-fake-1''dummy"), equalTo(""));
 	}
+
+	@Test
+	void testUrlBasicAuth() throws IOException {
+		URLConnection connection = new NoOpUrlConnection(new URL("https://SomeUser:${properties:src\\test\\resources\\url.properties::password}@example.com"));
```

Comment:
i'm not following what that syntax is meant to do? 

loading a relative file and grap a value from it?

Reply:
Yes, this syntax is defined in [org.apache.commons.text.StringSubstitutor](https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringSubstitutor.html). Anyway, as suggested in the first comment, I replaced it with PropertiesValueResolver.